sqlglot/parser.py

```diff
@@ -4484,6 +4491,47 @@ def _parse_as_command(self, start: Token) -> exp.Command:
         size = len(start.text)
         return exp.Command(this=text[:size], expression=text[size:])
 
+    def _parse_dict_property(self, this: str) -> exp.DictProperty:
+        settings = []
+
+        self._match(TokenType.L_PAREN)
+        kind = self._parse_id_var()
+        if not kind:
+            self.raise_error(f"Unexpected token: {self._curr.text}")
```

Comment:
self._curr could be none and you'll error out here

Reply:
Is it `None` on EOF?

Reply:
it's None, which means eof